File: src/main/java/io/github/Chase22/telegram/telegrambotapimock/api/endpoints/GetMeEndpoint.java
package io.github.Chase22.telegram.telegrambotapimock.api.endpoints;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.github.Chase22.telegram.telegrambotapimock.TelegramBotApiMock;
import io.github.Chase22.telegram.telegrambotapimock.infrastructure.server.TelegramApiEndpoint;
import io.undertow.server.HttpServerExchange;

import java.io.IOException;

public class GetMeEndpoint extends TelegramApiEndpoint<Void> {
    @Override
    public Class<Void> getBodyType() {
        return null;
    }

    @Override
    public String getPath() {
        return "/getMe";
    }

    @Override
    protected void process(final HttpServerExchange exchange) {
        try {
            new ObjectMapper().writeValue(exchange.getOutputStream(), TelegramBotApiMock.configuration.getBot());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/io/github/chase22/telegram/telegrambotapimock/telegram/endpoints/GetMeEndpoint.kt
package io.github.chase22.telegram.telegrambotapimock.telegram.endpoints

import com.fasterxml.jackson.databind.ObjectMapper
import io.github.chase22.telegram.telegrambotapimock.infrastructure.server.HttpExchangeWrapper
import io.github.chase22.telegram.telegrambotapimock.telegram.data.User
import java.io.IOException
import java.nio.ByteBuffer

class GetMeEndpoint(private val bot: User, private val objectMapper: ObjectMapper) : TelegramApiEndpoint<Nothing>() {
    override val bodyType: Nothing? = null

    override val path: String
        get() = "/getMe"

    override fun process(exchange: HttpExchangeWrapper) {
        try {
            exchange.send(objectMapper.writeValueAsBytes(bot))
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}
